### ISSUE TITLE
No external internet access when using containerd

### ISSUE THREAD

**Issue by mattdodge** [concourse/concourse#6960]
## Summary

Upgrading from guardian/runc to containerd and my containers do not have any external internet access. Not only DNS, but external IPs are failing. For example, I cannot ping `8.8.8.8` or `1.1.1.1` even.

My resolv.conf looks ok to me, but the problem doesn't even seem to be with DNS. I can't even reach external IP addresses. And iptables does not appear to be installed.

## Expected results
Expected to be able to ping external addresses like `8.8.8.8` or `google.com`.

## Actual results
```bash
$ ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8): 56 data bytes
^C
--- 8.8.8.8 ping statistics ---
179 packets transmitted, 0 packets received, 100% packet loss
$ ping 10.80.0.188
PING 10.80.0.188 (10.80.0.188): 56 data bytes
64 bytes from 10.80.0.188: seq=0 ttl=64 time=0.028 ms
64 bytes from 10.80.0.188: seq=1 ttl=64 time=0.060 ms
^C
--- 10.80.0.188 ping statistics ---
2 packets transmitted, 2 packets received, 0% packet loss
round-trip min/avg/max = 0.028/0.044/0.060 ms
```

## Additional context

Running native concourse binaries on EC2 machines. No docker or k8s involved.

**Concourse Worker Run Configuration**
```
/opt/concourse/bin/concourse worker \
  --tsa-host "10.35.xxx.xxx:2222" \
  --tsa-public-key "/opt/concourse/etc/host_key.pub" \
  --tsa-worker-private-key "/opt/concourse/etc/worker_key" \
  --work-dir "/opt/concourse/work" \
  --baggageclaim-driver "overlay" \
  --runtime "containerd" \
  --containerd-dns-server "8.8.8.8" \
  --containerd-cni-plugins-dir "/opt/concourse/bin" \
  --containerd-bin "/opt/concourse/bin/containerd" \
  --containerd-init-bin "/opt/concourse/bin/init" \
  `;
```

**Some more system info**
```bash
$ uname -a
Linux 2023fdd9-1e8a-44e9-5eb0-0dbca8559028 5.3.0-1017-aws #18~18.04.1-Ubuntu SMP Wed Apr 8 15:12:16 UTC 2020 x86_64 GNU/Linux
$ cat /etc/resolv.conf
nameserver 8.8.8.8
$ iptables -L
bash: iptables: command not found
```

## Triaging info

* Concourse version: 7.2.0
* Did this used to work?: Worked fine with guardian runtime, not containerd

**Comment by mattdodge** [comment]
Sorry, should have specified. Those commands were all run in the container. On the host I have the standard EC2 resolv.conf that is correctly being overwritten. The host has internet access too

**Comment by aoldershaw** [comment]
What does iptables -L give when run on the host?

**Comment by mattdodge** [comment]
iptables output from the host machine

```
Chain INPUT (policy ACCEPT)
target     prot opt source               destination
w--input   all  --  anywhere             anywhere

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination
CNI-FORWARD  all  --  anywhere             anywhere             /* CNI firewall plugin rules */
w--forward  all  --  anywhere             anywhere

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Chain CNI-FORWARD (1 references)
target     prot opt source               destination
CONCOURSE-OPERATOR  all  --  anywhere             anywhere             /* CNI firewall plugin admin overrides */
ACCEPT     all  --  anywhere             ip-10-80-0-2.us-west-2.compute.internal  ctstate RELATED,ESTABLISHED
ACCEPT     all  --  ip-10-80-0-2.us-west-2.compute.internal  anywhere

Chain CONCOURSE-OPERATOR (1 references)
target     prot opt source               destination
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED

Chain w--default (1 references)
target     prot opt source               destination
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED

Chain w--forward (1 references)
target     prot opt source               destination
ACCEPT     all  --  anywhere             anywhere
w--instance-uli99ifl8m0  all  --  ip-10-80-0-2.us-west-2.compute.internal  anywhere            [goto]  /* d3197ff7-5215-451c-651d-104902dcb773 */
DROP       all  --  anywhere             anywhere

Chain w--input (1 references)
target     prot opt source               destination
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
REJECT     all  --  anywhere             anywhere             reject-with icmp-host-prohibited

Chain w--instance-uli99ifl8m0 (1 references)
target     prot opt source               destination
ACCEPT     all  --  ip-10-80-0-0.us-west-2.compute.internal/30  ip-10-80-0-0.us-west-2.compute.internal/30  /* d3197ff7-5215-451c-651d-104902dcb773 */
w--default  all  --  anywhere             anywhere            [goto]  /* d3197ff7-5215-451c-651d-104902dcb773 */

Chain w--instance-uli99ifl8m0-log (0 references)
target     prot opt source               destination
LOG        all  --  anywhere             anywhere             ctstate INVALID,NEW,UNTRACKED /* d3197ff7-5215-451c-651d-104902dcb773 */ LOG level warning prefix "d3197ff7-5215-451c-651d-1049 "
RETURN     all  --  anywhere             anywhere             /* d3197ff7-5215-451c-651d-104902dcb773 */
```

**Comment by mattdodge** [comment]
Well weirdly this is now working fine on my test concourse server. It was failing for a while (failed to reach any network hosts) and now it's going. The test server is much smaller than our production one, maybe there's some switchover time from guardian to containerd I need to be more patient for? Let me try to upgrade production again this weekend and give it some more time to see if it starts working there too.

I see now that every new container seems to get it's own `ACCEPT` rule in the CNI-FORWARD chain of iptables, hopefully I'm interpreting that correctly. I'll be on the lookout for that as I upgrade. Is there an easy way to see a container's IP address short of hijacking and installing ifconfig?

**Comment by aoldershaw** [comment]
> Is there an easy way to see a container's IP address short of hijacking and installing ifconfig

Not really that I know of. You can get this info through the Garden API, but our containerd runtime doesn't expose that currently (though it's implemented by #6315) - but even going through the Garden API is kind of annoying, since you need to find the container handle first (through `fly containers`)

**Comment by muntac** [comment]
Going to close this as it was noted that the behaviour in question is now functional. Feel free to reopen if the issue persists.
